pandas/tests/extension/base/__init__.py

```diff
@@ -34,33 +34,61 @@ class TestMyDtype(BaseDtypeTests):
 wherever the test requires it. You're free to implement additional tests.
 
 """
-from pandas.tests.extension.base.accumulate import BaseAccumulateTests  # noqa: F401
-from pandas.tests.extension.base.casting import BaseCastingTests  # noqa: F401
-from pandas.tests.extension.base.constructors import BaseConstructorsTests  # noqa: F401
+from pandas.tests.extension.base.accumulate import BaseAccumulateTests
+from pandas.tests.extension.base.casting import BaseCastingTests
+from pandas.tests.extension.base.constructors import BaseConstructorsTests
 from pandas.tests.extension.base.dim2 import (  # noqa: F401
     Dim2CompatTests,
     NDArrayBacked2DTests,
 )
-from pandas.tests.extension.base.dtype import BaseDtypeTests  # noqa: F401
-from pandas.tests.extension.base.getitem import BaseGetitemTests  # noqa: F401
-from pandas.tests.extension.base.groupby import BaseGroupbyTests  # noqa: F401
-from pandas.tests.extension.base.index import BaseIndexTests  # noqa: F401
-from pandas.tests.extension.base.interface import BaseInterfaceTests  # noqa: F401
-from pandas.tests.extension.base.io import BaseParsingTests  # noqa: F401
-from pandas.tests.extension.base.methods import BaseMethodsTests  # noqa: F401
-from pandas.tests.extension.base.missing import BaseMissingTests  # noqa: F401
+from pandas.tests.extension.base.dtype import BaseDtypeTests
+from pandas.tests.extension.base.getitem import BaseGetitemTests
+from pandas.tests.extension.base.groupby import BaseGroupbyTests
+from pandas.tests.extension.base.index import BaseIndexTests
+from pandas.tests.extension.base.interface import BaseInterfaceTests
+from pandas.tests.extension.base.io import BaseParsingTests
+from pandas.tests.extension.base.methods import BaseMethodsTests
+from pandas.tests.extension.base.missing import BaseMissingTests
 from pandas.tests.extension.base.ops import (  # noqa: F401
     BaseArithmeticOpsTests,
     BaseComparisonOpsTests,
     BaseOpsUtil,
     BaseUnaryOpsTests,
 )
-from pandas.tests.extension.base.printing import BasePrintingTests  # noqa: F401
+from pandas.tests.extension.base.printing import BasePrintingTests
 from pandas.tests.extension.base.reduce import (  # noqa: F401
     BaseBooleanReduceTests,
     BaseNoReduceTests,
     BaseNumericReduceTests,
     BaseReduceTests,
 )
-from pandas.tests.extension.base.reshaping import BaseReshapingTests  # noqa: F401
-from pandas.tests.extension.base.setitem import BaseSetitemTests  # noqa: F401
+from pandas.tests.extension.base.reshaping import BaseReshapingTests
+from pandas.tests.extension.base.setitem import BaseSetitemTests
+
+
+# One test class that you can inherit as an alternative to inheriting all the
+# test classes above.
+# Note 1) this excludes Dim2CompatTests and NDArrayBacked2DTests.
+# Note 2) this uses BaseReduceTests and and _not_ BaseBooleanReduceTests,
+#  BaseNoReduceTests, or BaseNumericReduceTests
+class ExtensionTests(
+    BaseAccumulateTests,
```

Comment:
So the idea will be to remove each of these and fold each class' tests into `ExtensionTests`?

Reply:
yes